fortitude/src/rules/style/keywords.rs

```diff
@@ -0,0 +1,276 @@
+/// Defines rules that govern the use of keywords.
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{AstRule, FromAstNode};
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_source_file::SourceFile;
+use ruff_text_size::{TextRange, TextSize};
+use std::str::FromStr;
+use tree_sitter::Node;
+
+// TODO Support for `endfile`/`end file`
+
+#[derive(strum_macros::EnumString, strum_macros::Display)]
+#[strum(serialize_all = "lowercase", ascii_case_insensitive)]
```

Comment:
`ascii_case_insensitive` is the magic thing that preserves existing case here?

Reply:
No, in this case it just means `DoubleKeyword::from_str(text)` is case insensitive, and the combination with `serialize_all = "lowercase"` means `double_keyword.to_string()` is always lowercase. To preserve case in the fixes, I do it by just inserting a space/removing whitespace at the relevant offset. The error messages and fix titles both use lowercase regardless of what the user wrote. That could be corrected fairly easily though.

Reply:
Ah, yes, I see. No, that's fine